### PR TITLE
Llm improvements

### PR DESCRIPTION
**Pull request by maxime-louis**
What's been done:
* Max doc LEN paramenter add to config and check vllm uses it
* Rename vllm/LLM into vllm/VLLM
* VLLM: Sampling Temperature + beam search + best of with sampling (not tested yet)
* Refactor format_instruction
* Fix padding/truncation in LLM generator (and same for vllm)
* Check temperature for VLLM/greedy (should be 0 instead of 1) 
* System template not supported by gemma: handled
* Fix warnings from line 164 and line 171 of llm.py: removed
* update mistralai/Mistral-7B-Instruct-v0.2 config with the correct indent for batchsize  (RV/MAXIME)
    * for all models!
* Fixing ‘#Response’ usage (MAXIME) **This is the most tricky part of this PR, I checked that the new way to delimitate labels for FT is fine on llama31/mistral7B/gemma-2B/tinyLLAMA but there might be still some small edge cases... I put asserts in the code which will make the trainings fail if the label is not properly delimited**
* BERGEN Trainer: has been removed, logic is now simpler.
* target_modules changed to 'all-linear'

### CODE REVIEW

File: config/train/lora.yaml
Comment (AlexandreMisrahi2005) [inline config/train/lora.yaml:1]: maybe change ```test_size_ratio``` to ```test_size``` since it can support integer and float now

File: modules/rag.py [hunk 1]
```diff
@@ -447,30 +446,29 @@ def eval_metrics(self, dataset_split, questions, predictions, references):
...
-        from peft import LoraConfig, get_peft_model, prepare_model_for_kbit_training
-        from modules.trainer import RAGTrainer
         import torch
-        from transformers import Trainer
-        from modules.dataset import Tokenized_Sorted_Dataset
         from torch.utils.data import DataLoader
+        from transformers import TrainingArguments, Trainer
+        from peft import LoraConfig, get_peft_model, prepare_model_for_kbit_training
+        from modules.dataset import Tokenized_Sorted_Dataset
+        from modules.wandb_callback import WandbPredictionProgressCallback
```
Comment (sclincha) [inline modules/rag.py:454]: This import does not look necessary? Try,Catch around it to avoid dependencies to wandb?

Comment (maxime-louis) [reply modules/rag.py]: Solved

File: config/train/lora.yaml
Comment (maxime-louis) [reply config/train/lora.yaml:1]: Done

**Comment by sclincha** [comment]
Thanks a lot @maxime-louis !
